### ISSUE TITLE
ChefDK issue with aws-sdk gem

### ISSUE THREAD

**Issue by scalp42** [chef-cookbooks/aws#355]
Hi folks,

I'm not sure if the issue should be opened here on in ChefDK repo.

I've set up two kitchen suites with logs here: https://github.com/scalp42/chefdk-aws-issue

The example is a simple recipe installing `aws-sdk-kms` v3 and module requiring `aws-sdk-kms` (but any other aws gem will break I'm pretty sure):

```ruby
module Kitchen
  def self.test!
    require 'aws-sdk-kms'

    Chef::Log.info 'It works! YAY'
  end
end
```

If using the product `chef`, then everything will work as expected when the module method is called:

```
     Installing Cookbook Gems:
       Compiling Cookbooks...
       Recipe: chefdk-aws-issue::default
         * chef_gem[aws-sdk-kms] action install[2018-06-13T22:03:39+00:00] INFO: Processing chef_gem[aws-sdk-kms] action install (chefdk-aws-issue::default line 10)
       [2018-06-13T22:03:40+00:00] INFO: chef_gem[aws-sdk-kms] installed aws-sdk-kms at 1.5.0

           - install version 1.5.0 of package aws-sdk-kms
       [2018-06-13T22:03:41+00:00] INFO: It works! YAY
         Converging 1 resources
         * chef_gem[aws-sdk-kms] action install[2018-06-13T22:03:41+00:00] INFO: Processing chef_gem[aws-sdk-kms] action install (chefdk-aws-issue::default line 10)
        (up to date)
       [2018-06-13T22:03:41+00:00] INFO: Chef Run complete in 2.159324895 seconds
```

If using the product `chefdk`, then it'll error out with following error:

```
         Gem::ConflictError
         ------------------
         Unable to activate aws-sdk-kms-1.5.0, because aws-sdk-core-2.11.50 conflicts with aws-sdk-core (~> 3)

         Cookbook Trace:
         ---------------
           /var/tmp/kitchen/cache/cookbooks/chefdk-aws-issue/libraries/aws.rb:3:in `test!'
           /var/tmp/kitchen/cache/cookbooks/chefdk-aws-issue/recipes/default.rb:14:in `from_file'

         Relevant File Content:
         ----------------------
         /var/tmp/kitchen/cache/cookbooks/chefdk-aws-issue/libraries/aws.rb:

           1:  module Kitchen
           2:    def self.test!
           3>>     require 'aws-sdk-kms'
           4:
           5:      Chef::Log.info 'It works! YAY'
           6:    end
           7:  end
           8:

         System Info:
         ------------
         chef_version=14.1.12
         platform=ubuntu
         platform_version=16.04
         ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
         program_name=/opt/chefdk/bin/chef-client
         executable=/opt/chefdk/bin/chef-client
```

The only difference on fresh install is that `ChefDK` has `aws-sdk` v2 already installed:

```
root@default-ubuntu-1604:~# /opt/chefdk/bin/chef --version
Chef Development Kit Version: 3.0.36
chef-client version: 14.1.12
delivery version: master (7206afaf4cf29a17d2144bb39c55b7212cfafcc7)
berks version: 7.0.2
kitchen version: 1.21.2
inspec version: 2.1.72

root@default-ubuntu-1604:~# /opt/chefdk/embedded/bin/gem list | grep 'aws-sdk'
aws-sdk (2.11.50)
aws-sdk-core (2.11.50)
aws-sdk-resources (2.11.50)
```

It may or may not be related to https://github.com/chef-cookbooks/aws/issues/323.

TLDR: when requiring `aws-sdk-kms` v3, Chef works fine but ChefDK breaks.

Thanks for the help in advance 🤕

cc @tas50

**Comment by tas50** [comment]
Yep we have v2 already installed due to chef-provisioning. We need to get that onto v3 so we can move inspec to v3 and then v3 will work.

**Comment by bdwyertech** [comment]
Maybe we should not require the `aws-sdk` gem but instead depend only on the gems of cookbook-supported functions.  It seems the `aws-sdk` gem goes out and installs the whole encyclopedia of ruby gems for AWS services.

e.g.
```ruby
gem 'aws-sdk-ec2'
gem 'aws-sdk-ssm'
...
```

**Comment by majormoses** [comment]
This should be closed via #358 and released as `8.0.0`
